### PR TITLE
Added RedirectToPage overloads to Controller

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #6117 

Almost all of this is copy-paste and moving things around. I've looked through the doc comments to see if there are any wrong references but couldn't find any.

### CODE REVIEW

File: test/WebSites/RazorPagesWebSite/Controllers/RedirectController.cs [hunk 1]
```diff
@@ -8,9 +8,9 @@ namespace RazorPagesWebSite
     public class RedirectController : Controller
     {
         [HttpGet("/RedirectToPage")]
-        public IActionResult RedirectToPage()
+        public IActionResult RedirectToPageAction()
         {
-            return RedirectToRoute(new { page = "/RedirectToController", param = 17 });
+            return RedirectToPage("/RedirectToController", new { param = 17 });
```
Comment (ajaybhargavb) [inline test/WebSites/RazorPagesWebSite/Controllers/RedirectController.cs]: I just changed this one functional test for this. Are more tests needed?

File: src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs [hunk 2]
```diff
@@ -944,6 +944,155 @@ public virtual RedirectToRouteResult RedirectToRoutePermanent(string routeName,
...
         /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status302Found"/>) to the current page.
+        /// </summary>
+        /// <returns>The <see cref="RedirectToPageResult"/>.</returns>
+        [NonAction]
+        public virtual RedirectToPageResult RedirectToPage()
+            => RedirectToPage(pageName: null);
+
+        /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status302Found"/>) to the current page with the specified <paramref name="routeValues"/>.
```
Comment (ryanbrandenburg) [inline src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs]: > to the current page

If this overload is meant to redirect to the same page but with different `routeValues` we should leave it off of `ControllerBase` right?

File: src/Microsoft.AspNetCore.Mvc.Core/Resources.resx [hunk 3]
```diff
@@ -26,36 +26,36 @@
...
     extensible. For a given mimetype the value must be set accordingly:
-
-    Note - application/x-microsoft.net.object.binary.base64 is the format
-    that the ResXResourceWriter will generate, however the reader can
+    
+    Note - application/x-microsoft.net.object.binary.base64 is the format 
+    that the ResXResourceWriter will generate, however the reader can 
     read any of the formats listed below.
-
+
```
Comment (ryanbrandenburg) [inline src/Microsoft.AspNetCore.Mvc.Core/Resources.resx]: Weird that it messes these lines up when generating.

File: src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs [hunk 4]
```diff
@@ -944,6 +944,155 @@ public virtual RedirectToRouteResult RedirectToRoutePermanent(string routeName,
         }
 
         /// <summary>
+        /// Redirects (<see cref="StatusCodes.Status302Found"/>) to the current page.
+        /// </summary>
+        /// <returns>The <see cref="RedirectToPageResult"/>.</returns>
+        [NonAction]
+        public virtual RedirectToPageResult RedirectToPage()
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs]: This doesn't make sense here. Leave this one out.

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs]: Similarly the one below.

File: src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs [hunk 5]
```diff
@@ -944,6 +944,155 @@ public virtual RedirectToRouteResult RedirectToRoutePermanent(string routeName,
...
+        /// <see cref="RedirectToRouteResult.Permanent"/> set to false and <see cref="RedirectToRouteResult.PreserveMethod"/>
+        /// set to true, using the specified <paramref name="pageName"/>, <paramref name="routeValues"/>, and <paramref name="fragment"/>.
+        /// </summary>
+        /// <param name="pageName">The name of the page.</param>
+        /// <param name="routeValues">The route data to use for generating the URL.</param>
+        /// <param name="fragment">The fragment to add to the URL.</param>
+        /// <returns>The created <see cref="RedirectToRouteResult"/> for the response.</returns> 
+        [NonAction]
+        public virtual RedirectToPageResult RedirectToPagePreserveMethod(
+            string pageName = null,
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs]: Make this required. It really doesn't make sense to have a `null` page from a Controller.

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs [hunk 6]
```diff
@@ -230,6 +232,11 @@ static MvcCoreLoggerExtensions()
                 LogLevel.Debug,
                 3,
                 "No actions matched the current request. Route values: {RouteValues}");
+
+            _redirectToPageResultExecuting = LoggerMessage.Define<string>(
+                LogLevel.Information,
+                5,
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs]: Should this be `1`? Seems like that's the eventId for all the other redirect methods declared here.

**Comment by ajaybhargavb** [comment]
🆙 📅

**Pull request merged**